### PR TITLE
Add Maven rewrite support to update

### PR DESCRIPTION
**Pull request by ia3andy**
```
~/w/r/q/quarkus-logging-sentry> qss update --help                                                                                                                1.x!(origin/1.x)

Suggest recommended project updates with the possibility to apply them.

Usage: quarkus update [-eh] [--per-module] [--refresh] [--[no-]registry-client]
                      [--verbose] [--config=CONFIG] [-D=<String=String>]...
                      [-S=targetStream | -P=targetPlatformVersion]
                      [[--no-rewrite] | [--dry-run] |
                      --update-recipes-version=<updateRecipesVersion> |
                      --rewrite-plugin-version=<pluginVersion>]

Options:
  -e, --errors          Display error messages.
      --verbose         Verbose mode.
      --refresh         Refresh the local Quarkus extension registry cache
      --config=CONFIG   Configuration file
      --[no-]registry-client
                        Use the Quarkus extension catalog
  -h, --help            Display this help message.
  -D=<String=String>    Java properties
      --per-module      Display information per project module.

Target Quarkus version:
  -S, --stream=targetStream
                        A target stream, for example:
                          2.0
  -P, --platform-version=targetPlatformVersion
                        A specific target Quarkus platform version, for example:
                          2.2.0.Final


Rewrite:
      --no-rewrite      Disable the rewrite feature.
      --dry-run         Rewrite in dry-mode.
      --update-recipes-version=<updateRecipesVersion>
                        Use a custom io.quarkus:quarkus-update-recipes version.
                          This artifact contains the base recipes used by this
                          tool to update a project.
      --rewrite-plugin-version=<pluginVersion>
                        Use a custom OpenRewrite plugin version.
```

It works on extensions. If running directly from maven on a multi-module project, you need to add `-N` (non-recursive)
```
mvn io.quarkus:quarkus-maven-plugin:999-SNAPSHOT:update -N -Dstream=3.0
```

### CODE REVIEW

**Comment by ia3andy** [comment]
cc @maxandersen

**Comment by ia3andy** [comment]
@aloubyansky CI is passing we just need the review

**Comment by ia3andy** [comment]
@gsmet @aloubyansky @maxandersen up

**Comment by gsmet** [comment]
yes, I will work on that today.

File: independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java [hunk 1]
```diff
@@ -45,6 +45,7 @@ public class UpdateProjectCommandHandler implements QuarkusCommandHandler {
     public static final String REMOVE = "Remove:";
     public static final String UPDATE = "Update:";
     public static final String ITEM_FORMAT = "%-7s %s";
+    public static final String DEFAULT_UPDATE_RECIPES_VERSION = "LATEST";
```
Comment (gsmet) [inline independent-projects/tools/devtools-common/src/main/java/io/quarkus/devtools/commands/handlers/UpdateProjectCommandHandler.java:48]: Let's not hold this pull request for that but for your next work in this area, could you move this constant with the others in `QuarkusUpdateCommand`? I think it makes more sense to have them all in the same place.

**Review by gsmet — APPROVED** [review]
Let's get this in. It's a good first step, I tested it on my Quarkus GitHub App extension and it works as expected.

**Comment by ia3andy** [comment]
> Let's get this in. It's a good first step, I tested it on my Quarkus GitHub App extension and it works as expected.

This means we can start testing the recipe though the ecosystem-ci in [quarkus-updates/recipes at main · quarkusio/quarkus-updates](https://github.com/quarkusio/quarkus-updates) (I also made the versions configurable for this purpose)

**Pull request merged**